frontend/src/components/EmailSwapDialog.tsx

```diff
@@ -0,0 +1,249 @@
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle, Slide, TextField, Typography } from "@material-ui/core";
+import { Alert, AlertTitle } from "@mui/material";
+import React, { useContext } from "react";
+import { gql, useMutation } from '@apollo/client';
+import { useState } from "react";
+import { useFormik } from "formik";
+import ArrowBackIcon from '@mui/icons-material/ArrowBack';
+import CloseIcon from '@mui/icons-material/Close';
+import * as yup from 'yup'
+import router from "next/router";
+import { AuthContext } from "../providers/Auth";
+
+const SEND_SECONDARY_EMAIL_ACTIVATION = gql`
+mutation sendSecondaryEmailAvtivation($newEmail: String!, $password: String!){
+    sendSecondaryEmailActivation (email: $newEmail, password: $password){
+      success,
+      errors
+    }
+  }
+`
+
+const CONFIRM_EMAIL_CHANGE = gql`
+mutation confirmEmailChange($token: String!, $password: String!){
+    verifySecondaryEmail(
+      token: $token
+    ) {
+      success,
+      errors
+    }
+    swapEmails(
+      password: $password
+    ) {
+      success,
+      errors
+    }
+    removeSecondaryEmail(
+      password: $password
+    ) {
+      success,
+      errors
+    }
+  }
+`
+
+const newEmailFormInitialValues = {
+    newEmail: '',
+    password: ''
+}
+
+const newEmailFormValidationSchema = yup.object({
+    newEmail: yup
```

Comment:
the error validation is not being used in the form

Reply:
I am not really sure what you mean as I think that it is being used. I am merging anyway, lets discuss this another time maybe.